src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -59,25 +69,74 @@ public Task<ActionResult> GetDump(int? pid, [FromQuery]DumpType type = DumpType.
                     dumpFileName = $"core_{result.Pid}";
                 }
 
-                return CreateFileStreamResult(result, dumpFileName);
+                //Compression is done automatically by the response
+                //Chunking is done because the result has no content-length
+                return File(result.Value, "application/octet-stream", Invariant(dumpFileName));
             });
         }
 
         [HttpGet("cpuprofile/{pid?}")]
-        public Task<ActionResult> CpuProfile(int? pid, [FromQuery]int duration = 30)
+        public Task<ActionResult> CpuProfile(int? pid, [FromQuery]int durationSeconds = 30)
+        {
+            return InvokeService(async () =>
+            {
+                ValidateDuration(durationSeconds);
+                OperationResult<IStreamResult> result = await _diagnosticServices.StartCpuTrace(pid, durationSeconds, this.HttpContext.RequestAborted);
+                return new EventStreamResult(result.Value, "application/octet-stream", Invariant($"{Guid.NewGuid()}.nettrace"));
+            });
+        }
+
+        [HttpGet("trace/{pid?}")]
+        public Task<ActionResult> Trace(int? pid, [FromQuery]int durationSeconds = 30)
         {
             return InvokeService(async () =>
             {
-                OperationResult<Stream> result = await _diagnosticServices.StartCpuTrace(pid, duration, this.HttpContext.RequestAborted);
-                return CreateFileStreamResult(result, $"{Guid.NewGuid()}.nettrace");
+                ValidateDuration(durationSeconds);
+                OperationResult<IStreamResult> result = await _diagnosticServices.StartTrace(pid, durationSeconds, this.HttpContext.RequestAborted);
+                return new EventStreamResult(result.Value, "application/octet-stream", Invariant($"{Guid.NewGuid()}.nettrace"));
             });
         }
 
-        private FileStreamResult CreateFileStreamResult(OperationResult<Stream> result, FormattableString downloadName)
+        [HttpGet("logs/{pid?}")]
+        public ActionResult Logs(int? pid, [FromQuery] int durationSeconds = 30)
         {
-            //Compression is done automatically by the response
-            //Chunking is done because the result has no content-length
-            return File(result.Value, "application/octet-stream", Invariant(downloadName));
+            return InvokeService(() =>
+            {
+                ValidateDuration(durationSeconds);
+
+                return new OutputStreamResult(async (outputStream, token) =>
+                {
+                    await _diagnosticServices.StartLogs(outputStream, pid, durationSeconds, token);
+                }, "application/x-ndjson", Invariant($"{Guid.NewGuid()}.txt"));
+            });
+        }
+
+        private static void ValidateDuration(int durationSeconds)
+        {
+            if ((durationSeconds < 1) || (durationSeconds > MaxTraceSeconds))
```

Comment:
Why is there a max?

Reply:
PS this should be a 400 not 500.

Reply:
Ah I see the translation....

Reply:
We don't really have a stop command. The expectation is that you start the trace for a duration, and it ends automatically.

Reply:
I'm not a fan of the max duration though. It's fine if there's a default if nothing is specified but the default max seems like it should be removed.

Reply:
The browser experience for very long running /logs or /trace are not so great though. You have to wait until the end or the file is wiped. We don't really have a stop trace api or anything like that.

Reply:
(Note that the underlying implementation will support indefinite durations in cases where they are being used outside the rest server)

Reply:
@wiktork Thats why I think its OK to make the max the default, but not add an artifical max.